covenant-signer/signerapp/signer.go

```diff
@@ -138,3 +156,19 @@ func unbondSig(covenantPrivKey *btcec.PrivateKey, signingTxReq *ParsedSigningReq
 	}
 	return unbondingSig, nil
 }
+
+// stkExpSig signs the stake expansion transaction
+func stkExpSig(covenantPrivKey *btcec.PrivateKey, signingTxReq *ParsedSigningRequest) (*schnorr.Signature, error) {
+	stkExpSig, err := btcstaking.SignTxForFirstScriptSpendWithTwoInputsFromScript(
+		signingTxReq.StakingTx,
+		signingTxReq.StakeExp.PreviousActiveStakeTx.TxOut[signingTxReq.StakingOutputIdx],
+		signingTxReq.StakeExp.OtherFundingOutput,
+		covenantPrivKey,
+		signingTxReq.UnbondingScript,
```

Comment:
I think the `signingTxReq.UnbondingScript,` , is unbonding script of the current staking output i.e output from the expanded delegation. 

What we need need here is unbonding script from the staking output which funded stake expansion i.e the script commited in `signingTxReq.StakeExp.PreviousActiveStakeTx.TxOut[signingTxReq.StakingOutputIdx],`

Looking at Babylon code - https://github.com/babylonlabs-io/babylon/blob/main/x/btcstaking/keeper/msg_server.go#L469, we also need to fix it there.

To rebuild this script you can check how it is done in unbodning transaction - https://github.com/babylonlabs-io/babylon/blob/main/x/btcstaking/keeper/msg_server.go#L348

Reply:
I think after we have the https://github.com/babylonlabs-io/btc-staker/pull/188 ready, we can add e2e test in staker program which:
- sends the stake expansion request
- waits for it to be included in BTC

Such tests would caught this bug in Babylon node.

Reply:
updated the logic. Please review

Reply:
Test is failing currently cause need to update babylon node with the corresponding changes in https://github.com/babylonlabs-io/babylon/pull/1335